File: 1656.orderedStream.c
typedef struct {
    int size;
    int ptr;
    char** stream;
} OrderedStream;


OrderedStream* orderedStreamCreate(int n) {
    OrderedStream* new = malloc(sizeof(OrderedStream));

    new->size = n;
    new->ptr = 0;
    new->stream = malloc(n * sizeof(char*));

    for (size_t a = 0; a < n; ++a) {
        new->stream[a] = "";
    }

    return new;
}

char** orderedStreamInsert(OrderedStream* obj, int idKey, char* value, int* retSize) {
    /**
     * Adiciona 'value' no stream e se o 'idkye' do 'value' adicionado for equivalente
     * ao 'ptr', então retona uma lista de 'values' que sucedem o 'value' adicionado.
    */
    
    char** ret = NULL;

    *retSize = 0;
    obj->stream[idKey -1] = value;
    
    while (obj->ptr < obj->size && strcmp(obj->stream[obj->ptr], "")) {
        *retSize += 1;
        ret = realloc(ret, *retSize * sizeof(char*));
        ret[*retSize -1] = obj->stream[obj->ptr];
        ++obj->ptr;
    }
    return ret;
}

void orderedStreamFree(OrderedStream* obj) {
    
    free(obj->stream);
    obj->stream = NULL;

    free(obj);
    obj = NULL;
}

/**
 * Your OrderedStream struct will be instantiated and called as such:
 * OrderedStream* obj = orderedStreamCreate(n);
 * char** param_1 = orderedStreamInsert(obj, idKey, value, retSize);
 
 * orderedStreamFree(obj);
*/

void printret(char** list, int n) {
    if (n == 0) {
        puts("[]");
        return;
    }
    for (size_t a = 0; a < n; ++a) {
        puts(list[a]);
    }
}

File: 1656.orderedStream.py
class OrderedStream:

    def __init__(self, n: int):
        self.ptr = 0
        self.size = n
        self.stream = []

        for count in range(n):
            self.stream.append('')
        

    def insert(self, idKey: int, value: str) -> list[str]:
        self.stream[idKey -1] = value
        ret = []
        while self.ptr < self.size and self.stream[self.ptr] != "":
            ret.append(self.stream[self.ptr])
            self.ptr += 1
        return ret
        


# Your OrderedStream object will be instantiated and called as such:
# obj = OrderedStream(n)
# param_1 = obj.insert(idKey,value)
    
ob = OrderedStream(5)
param = ob.insert(3, 'ccccc')
print(param)
param = ob.insert(1, 'aaaaa')
print(param)
param = ob.insert(2, 'bbbbb')
print(param)
param = ob.insert(5, 'eeeee')
print(param)
param = ob.insert(4, 'ddddd')
print(param)